src/main/java/com/es/phoneshop/web/DeleteCartItemServlet.java

```diff
@@ -0,0 +1,24 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.model.cart.Cart;
+import com.es.phoneshop.model.cart.CartService;
+import com.es.phoneshop.model.cart.DefaultCartService;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class DeleteCartItemServlet extends HttpServlet {
+    CartService cartService = DefaultCartService.getInstance();
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String productId = req.getPathInfo().substring(1);
+        Cart cart = cartService.getCart(req);
+        cartService.delete(cart, Long.valueOf(productId));
```

Comment:
We still need to add validation on back-end: for the valid number ID existing product.
Otherwise, the user can alter HTML code to pass the arbitrary product ID in a method call.